### PR TITLE
feat: rehype plugin expand details and summary before rendering to epub

### PR DESCRIPTION
**Pull request by MateuszWasik**
[Closes 1051](https://github.com/unicorn-utterances/unicorn-utterances/issues/1051)

Adding new rehype plugin that will expand `<details>` before rendering to ePub

- transform `<details>` and `<summary>` into `divs` 
- apply `chevron down` unicode element to `<summary>` element as first children

image

### CODE REVIEW

**Comment by MateuszWasik** [comment]
Right now plugin is connected to `html` instead of `epub` just for testing purposes. `<details>` section is being expanded as expected.

File: src/utils/markdown/index.ts [hunk 1]
```diff
@@ -108,6 +109,7 @@ export function createRehypePlugins(config: MarkdownConfig): RehypePlugin[] {
 					rehypeUnicornElementMap,
 					rehypeTwoslashTabindex,
 					rehypeFileTree,
+					rehypeExpandDetailsAndSummary,
```
Comment (crutchcorn) [inline src/utils/markdown/index.ts:112]: Let's move this to the EPUB format, remove the `[WIP]` title, then we can merge

**Review by crutchcorn — CHANGES_REQUESTED** [review]
Minor change requested. Otherwise LGTM! :D

File: src/utils/markdown/index.ts (hunk 1 above)
Comment (MateuszWasik) [reply src/utils/markdown/index.ts:112]: It's in HTML format for a testing purposes. Unfortunately when it's moved to EPUB format it's not working as expected.

I tested it on macOS, ipadOS and on Kindle ebook reader. In those cases <details> elements are not expanded, even though this rehype plugin works correctly on HTML format.

Comment (MateuszWasik) [reply src/utils/markdown/index.ts:112]: I can't find a solution for this behaviour.

Comment (MateuszWasik) [reply src/utils/markdown/index.ts:112]: As discussed, we need to transform those elements into `divs` and add a new class that will style them to look like `<details>` elements

**Comment by MateuszWasik** [comment]
So actually `svg` elements is a miss or hit as @rjt-rockx says :D He actually proposed that we can use unicode to display `chevron` and that actually was a great idea. 

@crutchcorn @fennifith what do you think? Do we miss something by using unicode instead of svg element here?

File: src/utils/markdown/rehype-expand-details-summary.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import { visit } from "unist-util-visit";
+import { Element, Root } from "hast";
+
+const unicodeElement: Element = {
+	type: "element",
+	tagName: "span",
+	properties: {
+		style: "margin-right: 3px;",
```
Comment (MateuszWasik) [inline src/utils/markdown/rehype-expand-details-summary.ts:8]: We have to add some margin because `chevron` and `title` are too close to each other.

**Review by crutchcorn — APPROVED** [review]
Good work!

**Pull request merged**